From 8e8e354ba36e8a6aa2447f5492646ad205c14185 Mon Sep 17 00:00:00 2001
From: Thomas Thorne <Thomas.Thorne@Net2Edge.com>
Date: Tue, 20 Sep 2022 20:34:57 -0400
Subject: usb: serial: option: add 'reset_resume' callback for WWAN devices

This avoids the issue where the PPP modem does not reset correctly on resume,
causing issues with incoming calls on suspend.

Noted here: https://github.com/the-modem-distro/pinephone_modem_sdk/issues/97

This is part of a larger patch:
https://github.com/Quectel-Community/meta-quectel-community/commit/24adef2128f79b4d31490634438a917b3df462e2#diff-45dc5960ca2f1ef57139fe751587b3b043de9cca5864e2801210c00e75bd0780

However the rest of the patch is not needed/already upstreamed.
---
 drivers/usb/serial/option.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/serial/option.c b/drivers/usb/serial/option.c
index 8a5846d4adf6..65d7c2036fb8 100644
--- a/drivers/usb/serial/option.c
+++ b/drivers/usb/serial/option.c
@@ -2361,6 +2361,7 @@ static struct usb_serial_driver option_1port_device = {
 #ifdef CONFIG_PM
 	.suspend           = usb_wwan_suspend,
 	.resume            = usb_wwan_resume,
+	.reset_resume	   = usb_wwan_resume,
 #endif
 };
 
-- 
2.35.3

